File: solutions/python/P56_PowerfulDigitSum.py
# Project Euler Problem 56 Solution

from eutil.numbers import digit_list

def solve(A=100, B=100):
    best = 0
    for a in range(1, A):
        for b in range(0, B):
            best = max(sum(digit_list(a**b)), best)
    return best

if __name__ == "__main__":
    print("Solution:", solve())

File: solutions/haskell/P56_PowerfulDigitSum.hs
-- Project Euler Problem 56 Solution

import EulerUtil.Numbers (digitList)

-- solve :: Integral a => a -> a -> a
solve lim_a lim_b = foldl max 0 (map (sum.digitList) [ a^b | a<-[1..lim_a-1], b<-[0..lim_b-1]])

main = do
    putStrLn . show $ solve 100 100
